aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,170 @@
+
+To merely deactivate dynamic compression module, add a `<urlCompression>` element to your *web.config* file. There is no need to include an attribute and value for `doStaticCompression="false"`, since the IIS Static Compression Module doesn't work with ASP.NET Core applications in a reverse-proxy setup.
+```xml
+<configuration>
+  <system.webServer>
+    <urlCompression doDynamicCompression="false"/>
+  </system.webServer>
+</configuration>
+```
+If you opt to remove the module via *web.config*, you must unlock it first. Click on the IIS server in the IIS Manager **Connections** sidebar. Open the **Modules** in the IIS area. Click on the **DynamicCompressionModule** in the list. In the **Action** panel on the right, click **Unlock**. At this point, you will be able to add the section shown below to your *web.config* file to remove the module from the application. Doing this won't affect your use of the module in other applications on the server.
+```xml
+<configuration>
+  <system.webServer> 
+    <modules> 
+      <remove name="DynamicCompressionModule" /> 
+    </modules> 
+  </system.webServer> 
+</configuration>
+```
+
+## Troubleshooting
+Use a tool like [Fiddler](http://www.telerik.com/fiddler), [Firebug](http://getfirebug.com/), or [Postman](https://www.getpostman.com/), all of which allow you to set the `Accept-Encoding` request header and study the response headers, size, and body. The Response Compression Middleware will compress responses that meet the following conditions:
+* The `Accept-Encoding` header is present with a value of `gzip`, `*`, or custom encoding that matches a custom compression provider that you've established. The value must not be `identity` or have a quality value (qvalue, `q`) setting of 0 (zero).
+* The MIME type (`Content-Type`) must be set and must match a MIME type configured on the ResponseCompressionOptions.
+* The request must not include the `Content-Range` header.
+* The request must use *insecure protocol*, unless secure protocol is configured in the Response Compression Middleware options. *Note the danger [described above](#compression-with-secure-protocol) when enabling secure content compression.*
+* The provider must not use `GZipStream` on .NET Framework 4.5.1, which isn't flushable.
+
+## Additional Resources
+* [Application Startup](xref:fundamentals/startup)
+* [Middleware](xref:fundamentals/middleware)
+* [Apache Module mod_deflate](http://httpd.apache.org/docs/current/mod/mod_deflate.html)
+* [IIS HTTP Compression `<httpCompression>`](https://www.iis.net/configreference/system.webserver/httpcompression)
+* [NGINX Compression and Decompression](https://www.nginx.com/resources/admin-guide/compression-and-decompression/)
+* [Mozilla Developer Network: Accept-Encoding](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding)
+* [RFC 7231: Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4)
```

Comment:
We have linked to some of these previously already, not sure if they are worth repeating here.